Aleph 1 access module

cburch opened the pull request:
Pull request as tribute to the great Piggott Deity

cburch committed:
ALEPH-1 #comment initial access/security context work

cburch committed:
Merge remote-tracking branch 'origin/master' into ALEPH-1_access_module

Conflicts:
	aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/shared/ISecurityService.java

cburch committed:
Merge remote-tracking branch 'origin/master' into ALEPH-1_access_module

Conflicts:
	aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java

cburch committed:
Merge remote-tracking branch 'origin/master' into ALEPH-1_access_module

cburch committed:
ALEPH-1 ALEPH-2 Adding in stubs for isec and iaccess services as well as
a test project that should allow us to test plugging in whatever service
a person can create, added some sample tests that should give some code
coverage on the access, security, identity classes

cburch committed:
added the typesafe config lib and replaced my crazy temporary config
util

cburch committed:
ALEPH-1 ALEPH-2 removed all my prefix props files and switched over to
the typesafe config lib

cburch committed:
ALEPH-1 moved test case to not use injection

cburch committed:
ALEPH-1 added some fake services so we can test accessing them, need to
load them from a custom config

cburch committed:
ALEPH-1 removed the AccessDriver and moved it's code into ContextUtils,
cleaned up a bunch of the functions and tried to break them apart.
Removed IDataService and changed the AccessContext to just return the
requested dataservice via a passed in Class object instead.

cburch committed:
ALEPH-1 cleaned up some warnings

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

Conflicts:
	aleph2_access_manager/.classpath
	aleph2_access_manager/.gitignore
	aleph2_access_manager/.project
	aleph2_access_manager/.settings/org.eclipse.jdt.core.prefs
	aleph2_access_manager/pom.xml
	aleph2_data_model/pom.xml
	aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java
	aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessModule.java
	aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/shared_services/ISecurityService.java
	aleph2_data_model/src/com/ikanow/aleph2/data_model/utils/ContextUtils.java

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java:
```diff
@@ -30,9 +30,43 @@
+	public IManagementDbService getManagementDbService();
+	public ISearchIndexService getSearchIndexService();
+	public IStorageService getStorageIndexService();
+	public ITemporalService getTemporalService();
```
Hmm I can't decide whether we should keep these (and therefore add then to the Analytics/Harvest contexts, or keep the context interface more generic and just allow getDataService)?

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java:
```diff
@@ -30,9 +30,43 @@
+	//security service is related to data services
+	public ISecurityService getSecurityService();
+
+	public Injector getInjector();
```
What will this be used for?

cburch replied:
I need to add comments to all this, but the injector is used in that example I was sending around about using guice-servlet, you have to get the current injector and create a child from it with your apps injections so that you can inject the data services via @Inject tags

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java:
```diff
@@ -30,9 +30,43 @@
+ * @author Burch
+ *
+ */
+public interface IAccessContext {	
```
I don't understand ... my latest IAccessContext has an extra function call (for getting docs from an analytics/harveststream) that isn't in here ... hmm I think you need to do a pull before you can do this PR? (bitbucket handles this more nicely I think?)

cburch replied:
Your functions do exist, I merged before PR'ing, view whole file?  I think this only shows the diffs

Alex-Ikanow replied:
lol nvm I'm an idiot then

Alex Piggott
Director of Product Engineering | IKANOW
M: 443.310.5053
E:  apiggott@ikanow.com
W: www.ikanow.com


On Thu, May 7, 2015 at 11:24 AM, cburch <notifications@github.com> wrote:

> In
> aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java
> <https://github.com/IKANOW/Aleph2/pull/1#discussion_r29862783>:
>
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.IDocumentService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.IGeospatialService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.IGraphService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.IManagementDbService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.ISearchIndexService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.IStorageService;
> > +import com.ikanow.aleph2.data_model.interfaces.data_services.ITemporalService;
> > +import com.ikanow.aleph2.data_model.interfaces.shared_services.ISecurityService;
> > +
> > +/**
> > + * Gives an app access to all the currently configured data services.
> > + *
> > + * @author Burch
> > + *
> > + */
> > +public interface IAccessContext {	
>
> Your functions do exist, I merged before PR'ing, view whole file? I think
> this only shows the diffs
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IKANOW/Aleph2/pull/1/files#r29862783>.
>

Alex-Ikanow commented on aleph2_access_manager/test/com/ikanow/aleph2/access_manager/data_access/TestAccessManager.java:
```diff
@@ -0,0 +1,63 @@
+package com.ikanow.aleph2.access_manager.data_access;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_access_manager/test/com/ikanow/aleph2/access_manager/data_access/sample_services/SampleISecurityService.java:
```diff
@@ -0,0 +1,70 @@
+package com.ikanow.aleph2.access_manager.data_access.sample_services;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_access_manager/test/com/ikanow/aleph2/access_manager/data_access/sample_services/SampleUnboundService.java:
```diff
@@ -0,0 +1,7 @@
+package com.ikanow.aleph2.access_manager.data_access.sample_services;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_access_manager/test/com/ikanow/aleph2/access_manager/data_access/util/ConfigUtil.java:
```diff
@@ -0,0 +1,11 @@
+package com.ikanow.aleph2.access_manager.data_access.util;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java:
```diff
@@ -30,9 +30,43 @@
 /** A context library that is always passed to the AccessModules via injection 
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/data_access/IAccessContext.java:
```diff
@@ -30,9 +30,43 @@
 /** A context library that is always passed to the AccessModules via injection 
  * @author acp
```
change author :)

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/shared_services/ISecurityService.java:
```diff
@@ -15,15 +15,99 @@
  ******************************************************************************/
 package com.ikanow.aleph2.data_model.interfaces.shared_services;
 
+import java.util.Map;
```
Lots of @NonNulls or Optionals to be added here

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/interfaces/shared_services/ISecurityService.java:
```diff
@@ -15,15 +15,99 @@
+
 public interface ISecurityService {
+	
+	/**
```
These javadocs are a bit thin :)
What is the resource_class?

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/objects/shared/Identity.java:
```diff
@@ -0,0 +1,28 @@
+ * @author Burch
+ *
+ */
+public abstract class Identity {
```
Why an abstract class instead of an interface?

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/objects/shared/Identity.java:
```diff
@@ -0,0 +1,28 @@
+package com.ikanow.aleph2.data_model.objects.shared;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_data_model/src/com/ikanow/aleph2/data_model/utils/ContextUtils.java:
```diff
@@ -15,12 +15,32 @@
+import com.typesafe.config.ConfigFactory;
 
-public class ContextUtils {
+public class ContextUtils extends AbstractModule  {
```
hmm not sure about this.... XxxUtils classes should be containers of static functions, seems like this has become a little bit more...

Alex-Ikanow commented on aleph2_test_data_services/test/com/ikanow/aleph2/data_model/interfaces/data_access/TestIAccessContext.java:
```diff
@@ -0,0 +1,44 @@
+package com.ikanow.aleph2.data_model.interfaces.data_access;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_test_data_services/test/com/ikanow/aleph2/data_model/interfaces/shared_services/TestISecurityService.java:
```diff
@@ -0,0 +1,188 @@
+package com.ikanow.aleph2.data_model.interfaces.shared_services;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_test_data_services/test/com/ikanow/aleph2/data_model/objects/shared/TestIdentity.java:
```diff
@@ -0,0 +1,59 @@
+package com.ikanow.aleph2.data_model.objects.shared;
```
Copyleft header (Apache - it's on the github wiki)

Alex-Ikanow commented on aleph2_test_data_services/test/com/ikanow/aleph2/test_utils/LoadConfigUtility.java:
```diff
@@ -0,0 +1,18 @@
+package com.ikanow.aleph2.test_utils;
```
Copyleft header (Apache - it's on the github wiki)

cburch committed:
ALEPH-1 moved the config loading logic out of ContextUtils, added the
copyright messaging everywhere, added @NonNull to method headers where
necessary and added additional documentation

cburch committed:
mid changing over iaccessmodule being app centric

cburch committed:
asdf

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

cburch committed:
ALEPH-1 #comment fixing git so I can merge

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

Conflicts:
	aleph2_core_distributed_services_library/.settings/org.eclipse.jdt.core.prefs
	aleph2_data_analytics_manager/.settings/org.eclipse.jdt.core.prefs
	aleph2_data_import_manager/.settings/org.eclipse.jdt.core.prefs
	aleph2_data_model/.settings/org.eclipse.jdt.core.prefs
	aleph2_data_model/pom.xml

cburch committed:
ALEPH-1 @comment added in new service context interface to handle
getting services instead of the access context

cburch committed:
ALEPH-1 @comment moved around the initialization and added some comments

cburch committed:
ALEPH-1 #comment moved all the iservice code into data_model under
moduleutils

cburch committed:
Merge remote-tracking branch 'Aleph2/master' into ALEPH-1_access_module

Conflicts:
	aleph2_access_manager/.classpath
	aleph2_core_distributed_services_library/.classpath
	aleph2_data_analytics_manager/.classpath
	aleph2_data_import_manager/.classpath
	aleph2_data_model/pom.xml

cburch committed:
ALEPH-1 #comment copy classpath in plus add junit

cburch committed:
ALEPH-1 #comment removed errors in test code

cburch committed:
ALEPH-1 #comment made test cases work so build will succeed

cburch committed:
ALEPH-1 #comment I didn't write these tests yet
